src/lib/auth.ts

```diff
@@ -93,10 +92,10 @@ export const DEVICE_API_KEY_PERMISSIONS = [
 
 	'resin.service.read?application/canAccess()',
 
-	'resin.service_install.read?installs__service/canAccess()',
+	'resin.service_install.read?device/canAccess()',
 	...writePerms(
 		'resin.service_install',
-		`installs__service/any(s:s/${applicationControlsDevice})`,
+		`device/any(d:d/${matchesActor} or d/belongs_to__application/any(a:a/depends_on__application/any(da:da/${ownsDevice})))`,
```

Comment:
This again granted write access to all service_installs of all device of the fleet.